lmdeploy/turbomind/deploy/source_model/qwen.py

```diff
@@ -112,3 +112,53 @@ def model_info(self):
                     max_position_embeddings=seq_length,
                     use_dynamic_ntk=int(use_dynamic_ntk),
                     use_logn_attn=use_logn_attn)
+
+
+class Qwen2Reader(LlamaReader):
+    """read qwen2 model weights.
+
+    The weight name of qwen2 model is similar to llama, except its attention
+    bias doesn't include o_proj bias. Therefore, we make a dummy zero o_proj
+    bias to make it comply the definition of turbomind llama format
+    """
+
+    def __init__(self, new_params: dict, unused_params: dict, last_bin: bool,
+                 model_cfg: dict):
+        super().__init__(new_params, unused_params, last_bin, model_cfg)
+
+    def attn_bias(self, i: int):
+        """Get q, k, v bias for layer i."""
+        result = []
+
+        for key in ['q', 'k', 'v']:
+            tensor = self.params.get(
+                f'model.layers.{i}.self_attn.{key}_proj.bias')
+            assert tensor is not None
+            result.append(tensor)
+
+        tensor = self.params.get(f'model.layers.{i}.self_attn.o_proj.weight')
+        dummy_oproj_bias = torch.zeros(tensor.shape[-1],
```

Comment:
maybe we could use `tensor.new_zeros(tensor.shape[-1])` to create the dumpy bias with same `dtype` and `device` as `tensor`